mmv1/products/cloudquotas/QuotaPreference.yaml

```diff
@@ -0,0 +1,164 @@
+  - !ruby/object:Api::Type::String
+    name: 'service'
+    required: true
+    description: The name of the service to which the quota preference is applied.
+  - !ruby/object:Api::Type::String
+    name: 'quotaId'
+    required: true
+    description: |
+      The id of the quota to which the quota preference is applied. A quota id is unique in the service.
+      Example: CPUS-per-project-region.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'quotaConfig'
+    required: true
+    description: The preferred quota configuration.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'preferredValue'
+        required: true
+        description: |
+          The preferred value. Must be greater than or equal to -1. If set to -1, it means the value is "unlimited".
+      - !ruby/object:Api::Type::String
+        name: 'stateDetail'
+        output: true
+        description: Optional details about the state of this quota preference.
+      - !ruby/object:Api::Type::String
+        name: 'grantedValue'
+        output: true
+        description: Granted quota value.
+      - !ruby/object:Api::Type::String
+        name: 'traceId'
+        output: true
+        description: |
+          The trace id that the Google Cloud uses to provision the requested quota. This trace id may be used by the client to contact Cloud support to track the state of a quota preference request. The trace id is only produced for increase requests and is unique for each request. The quota decrease requests do not have a trace id.
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'annotations'
+        description: |-
+          The annotations map for clients to store small amounts of arbitrary data. Do not put PII or other sensitive information here. See https://google.aip.dev/128#annotations.
+
+          An object containing a list of "key: value" pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+      - !ruby/object:Api::Type::Enum
+        name: 'requestOrigin'
+        output: true
+        description: The origin of the quota preference request.
+        values:
+          - :ORIGIN_UNSPECIFIED
+          - :CLOUD_CONSOLE`
+          - :AUTO_ADJUSTER
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'dimensions'
+    description: |-
+      The dimensions that this quota preference applies to. The key of the map entry is the name of a dimension, such as "region", "zone", "network_id", and the value of the map entry is the dimension value. If a dimension is missing from the map of dimensions, the quota preference applies to all the dimension values except for those that have other quota preferences configured for the specific value.
+
+      NOTE: QuotaPreferences can only be applied across all values of "user" and "resource" dimension. Do not set values for "user" or "resource" in the dimension map.
+
+      Example: {"provider", "Foo Inc"} where "provider" is a service specific dimension.
```

Comment:
Is there a typo here? Should it be `{"provider": "Foo Inc"}`?

Reply:
Yes fixed.